frontend/src/libraries/explorer-wamp/transactions.ts

```diff
@@ -435,24 +463,90 @@ export default class TransactionsApi extends ExplorerApi {
+          (action: RpcAction | string) => {
+            if (typeof action === "string") {
+              return { kind: action, args: {} };
+            } else {
+              const kind = Object.keys(action)[0] as keyof RpcAction;
+              return {
+                kind,
+                args: action[kind],
+              };
+            }
+          }
+        );
+
+        let receipts = transactionExtraInfo.receipts as ReceiptsList[];
+        const receiptsOutcome = transactionExtraInfo.receipts_outcome as ReceiptOutcome[];
+        if (
+          receipts.length === 0 ||
+          receipts[0].receipt_id !== receiptsOutcome[0].id
+        ) {
+          receipts.unshift({
+            predecessor_id: transactionExtraInfo.transaction.signer_id,
+            receipt: actions,
+            receipt_id: receiptsOutcome[0].id,
+            receiver_id: transactionExtraInfo.transaction.receiver_id,
+          });
+        }
+        const receiptOutcomesByIdMap = new Map();
+        receiptsOutcome.forEach((receipt: any) => {
+          receiptOutcomesByIdMap.set(receipt.id, [receipt]);
+        });
+
+        const receiptsByIdMap = new Map();
+        receipts.forEach((receiptItem: any) => {
+          if (receiptItem.receipt_id === receiptsOutcome[0].id) {
+            receiptItem.actions = actions;
           } else {
-            const kind = Object.keys(action)[0] as keyof RpcAction;
-            return {
-              kind,
-              args: action[kind],
-            };
+            const { Action: action = undefined } = receiptItem.receipt;
+            receiptItem.actions = action?.actions.map(
+              (action: RpcAction | string) => {
+                if (typeof action === "string") {
+                  return { kind: action, args: {} };
+                } else {
+                  const kind = Object.keys(action)[0] as keyof RpcAction;
+                  return {
+                    kind,
+                    args: action[kind],
+                  };
+                }
+              }
+            );
           }
+          receiptsByIdMap.set(receiptItem.receipt_id, [receiptItem]);
         });
 
-        transactionInfo.receiptsOutcome = transactionExtraInfo.receipts_outcome as ReceiptOutcome[];
+        const executionOutcomeReceipt = (receiptHash: string) => {
+          const receipt = receiptsByIdMap.get(receiptHash)[0];
+          const receiptOutcome = receiptOutcomesByIdMap.get(receiptHash)[0];
+          const receipts = { ...receipt, ...receiptOutcome };
+          return {
+            ...receipts,
+            outcome: {
+              ...receipts.outcome,
+              receipt_ids:
+                receipts.outcome && receipts.outcome.receipt_ids.length > 0
+                  ? receipts.outcome.receipt_ids.map(
+                      (executedReceipt: string) =>
+                        executionOutcomeReceipt(executedReceipt)
+                    )
+                  : [],
+            },
```

Comment:
`receiptOutcome.outcome` cannot be missing, so the following should work fine:

```suggestion
          const receipt = receiptsByIdMap.get(receiptHash);
          const receiptOutcome = receiptOutcomesByIdMap.get(receiptHash);
          return {
            ...receipt,
            ...receiptOutcome,
            outcome: {
              ...receiptOutcome.outcome,
              outgoing_receipts:
                receiptOutcome.outcome.receipt_ids.map(
                      (outgoingReceiptId: string) =>
                        executionOutcomeReceipt(outgoingReceiptId)
                    ),
            },
```